consensus/fork_choice/tests/tests.rs

```diff
@@ -1292,3 +1295,667 @@ async fn progressive_balances_cache_proposer_slashing() {
+        let fast_confirmed_head = test
+            .harness
+            .chain
+            .canonical_head
+            .fork_choice_read_lock()
+            .get_fast_confirmed_head();
+        assert!(fast_confirmed_head.is_none());
+    }
+
+    // Test with FCR enabled
+    #[cfg(feature = "fast_confirmation")]
+    {
+        let chain_config = ChainConfig {
+            fast_confirmation_enabled: true,
+            fcr_byzantine_threshold_basis_points: 2500,
+            ..ChainConfig::default()
+        };
+
+        let test = ForkChoiceTest::new_with_chain_config(chain_config);
+
+        // Apply blocks and verify normal fork choice still works
+        let test = test
+            .apply_blocks_while(|_, state| state.finalized_checkpoint().epoch == 0)
+            .await
+            .unwrap()
+            .apply_blocks(1)
+            .await;
+
+        // Verify normal fork choice functionality
+        let head = test.harness.head_block_root();
+        assert!(!head.is_zero());
+
+        let finalized_epoch = test
+            .harness
+            .chain
+            .canonical_head
+            .cached_head()
+            .finalized_checkpoint()
+            .epoch;
+        assert_eq!(finalized_epoch, 2);
+
+        let justified_epoch = test
+            .harness
+            .chain
+            .canonical_head
+            .cached_head()
+            .justified_checkpoint()
+            .epoch;
+        assert!(justified_epoch >= 1);
+
+        // Verify FCR also works
+        let fast_confirmed_head = test
+            .harness
+            .chain
+            .canonical_head
+            .fork_choice_read_lock()
+            .get_fast_confirmed_head();
+        assert!(fast_confirmed_head.is_some());
+    }
+}
+
+/// Tests that FCR works correctly with the feature flag disabled
+/// TODO: This test will need updates when FCR implementation is complete
+#[tokio::test]
+async fn fcr_feature_disabled_behavior() {
+    // Test with FCR enabled in config but feature flag disabled
+    let chain_config = ChainConfig {
+        fast_confirmation_enabled: true,
+        fcr_byzantine_threshold_basis_points: 2500,
+        ..ChainConfig::default()
+    };
+
+    let test = ForkChoiceTest::new_with_chain_config(chain_config);
+    let test = test
+        .apply_blocks_while(|_, state| state.finalized_checkpoint().epoch == 0)
+        .await
+        .unwrap()
+        .apply_blocks(1)
+        .await;
+
+    // When feature flag is disabled, FCR should return None regardless of config
+    #[cfg(not(feature = "fast_confirmation"))]
+    {
+        let fast_confirmed_head = test
+            .harness
+            .chain
+            .canonical_head
+            .fork_choice_read_lock()
+            .get_fast_confirmed_head();
+        assert!(fast_confirmed_head.is_none());
+    }
+
+    // Normal fork choice should still work
+    let head = test.harness.head_block_root();
+    assert!(!head.is_zero());
+}
```

Comment:
Didn't read this whole file but the tests seem to be heading in a good direction!